appinventor/components/src/com/google/appinventor/components/runtime/Notifier.java

```diff
@@ -217,8 +217,13 @@ public void ShowChooseDialog(String message, String title, final String button1T
         cancelable,
         new Runnable() {public void run() {AfterChoosing(button1Text);}},
         new Runnable() {public void run() {AfterChoosing(button2Text);}},
-        new Runnable() {public void run() {AfterChoosing(activity.getString(android.R.string.cancel));}}
-        );
+        new Runnable() {
+          public void run() {
+            AfterChoosing(activity.getString(android.R.string.cancel)); // backward compatible
+            ChoosingCanceled();
```

Comment:
I think that we should swap the order of these calls so that the more specific case (`ChoosingCanceled`) is run before the more generic case (`AfterChoosing(text)`).